File: test-agent.js
#!/usr/bin/env node

/**
 * Test script for the Research Agent
 * Run with: node test-agent.js
 */

import { runResearchAgentWorkflow } from './lib/researchAgent.js';
import dotenv from 'dotenv';

// Load environment variables from .env file
dotenv.config();

async function testAgent() {
  console.log('🚀 Testing AI Legislation Research Agent...\n');

  if (!process.env.OPENAI_API_KEY) {
    console.error('❌ Error: OPENAI_API_KEY not found in environment variables');
    console.error('Please create a .env file with your OpenAI API key\n');
    process.exit(1);
  }

  try {
    console.log('📝 Running query: "Generate the latest AI legislation brief for the past month"\n');
    
    const result = await runResearchAgentWorkflow({
      input_as_text: "Generate the latest AI legislation brief for the past month"
    });

    console.log('✅ Agent execution completed!\n');
    console.log('📊 Parsed Output:\n');
    console.log(JSON.stringify(result.output_parsed, null, 2));
    console.log('\n📄 Text Output:\n');
    console.log(result.output_text);

  } catch (error) {
    console.error('❌ Error running agent:', error.message);
    if (error.stack) {
      console.error('\nStack trace:', error.stack);
    }
    process.exit(1);
  }
}

// Run the test
testAgent();

File: test_agent.py
#!/usr/bin/env python3
"""
Test script for the Research Agent
Run with: python test_agent.py
"""

import os
import sys
import json
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Add lib to path
sys.path.insert(0, os.path.dirname(__file__))

from lib.research_agent import run_research_agent


def test_agent():
    print('🚀 Testing AI Legislation Research Agent...\n')
    
    if not os.getenv('OPENAI_API_KEY'):
        print('❌ Error: OPENAI_API_KEY not found in environment variables')
        print('Please create a .env file with your OpenAI API key\n')
        sys.exit(1)
    
    try:
        print('📝 Running query: "Generate the latest AI legislation brief for the past month"\n')
        
        result = run_research_agent(
            "Generate the latest AI legislation brief for the past month"
        )
        
        print('✅ Agent execution completed!\n')
        print('📊 Agent JSON Output:\n')
        print(json.dumps(result, indent=2))
        print('\n' + '='*80 + '\n')
        
    except Exception as e:
        print(f'❌ Error running agent: {str(e)}')
        import traceback
        traceback.print_exc()
        sys.exit(1)


if __name__ == '__main__':
    test_agent()

